go-controller/pkg/ovn/controller/apbroute/network_client.go

```diff
@@ -155,6 +179,13 @@ func (nb *northBoundClient) addGatewayIPs(pod *v1.Pod, egress gatewayInfoList) e
 	if util.PodCompleted(pod) || util.PodWantsHostNetwork(pod) {
 		return nil
 	}
+	local, err := nb.isPodInLocalZone(pod)
```

Comment:
may want to check if config.OVNKubernetesFeature.EnableInterconnect everywhere you check isPodInLocalZone